### PR TITLE
Add suspend/resume engine to the api

### PR DESCRIPTION
**Pull request by Akrion**
- Added `suspend` & `resume` to the engines api
- Added examples
- Updated change log and versions

### CODE REVIEW

**Review by meruyert93 — APPROVED** [review]
nice! codewise looks good to me 👍🏻

File: railib/api.py [hunk 1]
```diff
@@ -373,6 +375,30 @@ def create_engine_wait(ctx: Context, engine: str, size: str = "XS", **kwargs):
...
 
+def suspend_engine(ctx: Context, engine: str, **kwargs):
+    data = { "suspend": True }
+    url = _mkurl(ctx, f"{PATH_ENGINE}/{engine}")
+    rsp = rest.patch(ctx, url, data, **kwargs)
+    return json.loads(rsp.read())
+
+
+def resume_engine(ctx: Context, engine: str, **kwargs):
+    data = { "suspend": False }
```
Comment (vilterp) [inline railib/api.py:386]:
```suggestion
    data = {"suspend": False}
```

File: railib/api.py [hunk 2]
```diff
@@ -373,6 +375,30 @@ def create_engine_wait(ctx: Context, engine: str, size: str = "XS", **kwargs):
     return get_engine(ctx, engine)
 
 
+def suspend_engine(ctx: Context, engine: str, **kwargs):
+    data = { "suspend": True }
```
Comment (vilterp) [inline railib/api.py:379]:
```suggestion
    data = {"suspend": True}
```

File: examples/suspend_engine.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+# Answers if the given state is a terminal state.
+def is_term_state(state: str) -> bool:
+    return state == "SUSPENDED" or ("FAILED" in state)
+
+
+def run(engine: str, profile: str):
+    cfg = config.read(profile=profile)
+    ctx = api.Context(**cfg)
+    rsp = api.suspend_engine(ctx, engine)
```
Comment (vilterp) [inline examples/suspend_engine.py:32]: do we want a `suspend_engine_wait` as well or nah?

**Review by vilterp — COMMENTED** [review]
I wish engines would unsuspend themselves when you run queries against them, like Snowflake's virtual warehouses do. But in the meantime, might as well

**Review by vilterp — APPROVED** [review]

**Pull request merged**